frontend/src/hooks/useDashboardData.js

```diff
@@ -0,0 +1,91 @@
+import { useQuery } from '@tanstack/react-query';
+import { useState } from 'react';
+import { axiosInstance } from 'utils/axios';
+import { ZETH_ADDRESS } from 'utils/constants';
+import { ReactComponent as CollateralIcon } from 'assets/icons/collateral.svg';
+import { ReactComponent as EthIcon } from 'assets/icons/ethereum.svg';
+import { ReactComponent as UsdIcon } from 'assets/icons/usd_coin.svg';
+import { ReactComponent as BorrowIcon } from 'assets/icons/borrow.svg';
+import { ReactComponent as StrkIcon } from 'assets/icons/strk.svg';
+
+const fetchCardData = async (walletId) => {
+  if (!walletId) {
+    console.error('fetchCardData: walletId is undefined');
+    return null;
+  }
+  try {
+    const response = await axiosInstance.get(`/api/dashboard?wallet_id=${walletId}`);
+    return response.data;
+  } catch (error) {
+    console.error('Error during getting the data from API', error);
+    return null;
+  }
+};
+
+export const useDashboardData = (walletId) => {
+  const [cardData, setCardData] = useState([
+    {
+      title: 'Collateral & Earnings',
+      icon: CollateralIcon,
+      balance: '0.00',
+      currencyName: 'Ethereum',
+      currencyIcon: EthIcon,
+    },
+    {
+      title: 'Borrow',
+      icon: BorrowIcon,
+      balance: '0.00',
+      currencyName: 'USD Coin',
+      currencyIcon: UsdIcon,
+    },
+  ]);
+
+  const [healthFactor, setHealthFactor] = useState('0.00');
+  const [currentSum, setCurrentSum] = useState(0);
+
+  const { isLoading } = useQuery({
+    queryKey: ['dashboardData', walletId],
+    queryFn: () => fetchCardData(walletId),
+    enabled: !!walletId,
+    onSuccess: (data) => {
```

Comment:
onSuccess won't work and should be replaced with useEffect, that check If useQuery `isSuccess` -> yes -> put whole "onSuccess" functionality inside